Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/StatusBean.java

```diff
@@ -52,132 +60,180 @@
-
-        HashMap<String, Long> map = new HashMap<>();
-        for (EventCount event : snapshot.getEvents().values()) {
-            map.put(event.getEvent(), event.getOutbound());
-        }
-        return map;
-    }
-
-    public long getTotalOutboundRequests() {
-        long total = 0;
-        for (EventCount event : snapshot.getEvents().values()) {
-            total += event.getOutbound();
-        }
-        return total;
-    }
-
-    public List<AvailableService> getServices() {
-        return services;
-    }
-
-    public HorizontalBarChartModel getEventBarChart() {
-        return eventBarChart;
-    }
-
-    public PieChartModel getEventPieChart() {
-        if (eventPieChart != null) {
-            refreshPieChart();
-        }
-        return eventPieChart;
-    }
-
-    private void refreshPieChart() {
-
-        for ( EventCount event : snapshot.getEvents().values()) {
-            eventPieChart.getData().put(event.getEvent(), event.getTotal());
-        }
-
-        eventPieChart.setFill(false);
-        eventPieChart.setSeriesColors("10253F, CC0000, 33D6FF, FFCC00, 98FB98, FFA500, 00CCFF");
-        eventPieChart.setShowDataLabels(true);
-        eventPieChart.setSliceMargin(5);
-        eventPieChart.setLegendPosition("se");
-    }
-
-    private void initBarChart() {
-        eventBarChart = new HorizontalBarChartModel();
-
-        ChartSeries inboundSeries = createChart("Inbound", getInboundEventCounts());
-        ChartSeries outboundSeries = createChart("Outbound", getOutboundEventCounts());
-
-        eventBarChart.addSeries(inboundSeries);
-        eventBarChart.addSeries(outboundSeries);
-        eventBarChart.setSeriesColors("10253F, CC0000");
-        eventBarChart.setStacked(true);
-        eventBarChart.setLegendPosition("se");
-    }
-
-    private static ChartSeries createChart(String label, Map<String, Long> eventCounts) {
-        ChartSeries series = new ChartSeries();
-        series.setLabel(label);
-        for (Entry<String, Long> serviceEntry : eventCounts.entrySet()) {
-            series.set(serviceEntry.getKey(), serviceEntry.getValue());
-        }
-        return series;
-    }
-
-    private void initPieChart() {
-        eventPieChart = new PieChartModel();
-        refreshPieChart();
-    }
+	private final DashboardStatusServiceImpl sService = new DashboardStatusServiceImpl();
+	private final PingService pingService = new PingServiceImpl();
+	private List<AvailableService> services = new ArrayList<>();
+	private HorizontalBarChartModel eventBarChart;
+	private PieChartModel eventPieChart;
+	private StatusSnapshot snapshot;
+	private final CertificateManagerService certificateService;
+
+	private static final String KEY_STORE_MSG = "keyStoreMsg";
+	private static final Logger LOG = LoggerFactory.getLogger(StatusBean.class);
+	public List<CertificateDTO> certDTOs = new ArrayList<>();
+
+	public StatusBean() {
+		certificateService = new CertificateManagerServiceImpl();
```

Comment:
Optional: don't need a constructor here, you can use the @PostConstruct method to achieve the same result

Reply:
My personal opinion, but I'd rather it be initialized on line 69 like the rest of the services here, and getCertificateInfo() called in the initServices() method.